docs/models.rst

```diff
@@ -0,0 +1,220 @@
+Data Models
+===========
+
+The Rossum API SDK provides type-safe dataclass models for all API resources.
+These models are automatically populated from API responses and provide
+clear type hints for all fields.
+
+.. contents:: Table of Contents
+   :local:
+   :depth: 1
+
+Overview
+--------
+
+All models are Python dataclasses with full type annotations. They represent
+the structure of data returned by the Rossum API and can also be used when
+sending data to the API.
+
+**Key features:**
+
+* Type-safe with full type hints
+* Automatic deserialization from API responses
+* Support for nested objects
+* Optional fields with sensible defaults
+* Direct mapping to API documentation
```

Comment:
I would add a note somewhere in this section that you are able to define a custom deserializer to deserialize into your own models should you need it.